PricingLoader.scala

```diff
@@ -0,0 +1,54 @@
+package org.llm4s
+
+import scala.io.Source
+import scala.util.{Try, Using}
+import play.api.libs.json._
+
+object PricingLoader {
+  private val pricingUrl = "https://raw.githubusercontent.com/BerriAI/litellm/main/model_prices_and_context_window.json"
+  private var cachedPricing: Option[JsValue] = None
+
+  // Load pricing data from URL
+  def loadPricing(): Option[JsValue] = {

```

Comment:
Can we take all of litellms logic here:

https://github.com/BerriAI/litellm/blob/d468a2fa3483bbcd8adbe7e4a0df211734447099/litellm/__init__.py#L262

1. Allow for local override
2. fetch from the url
3. Fallback to a local copy of the file.
In each case we should add good logging to explain what we have done.